ccf_transaction_processor/transaction_processor/verify_signatures.cpp

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "pdo_tp.h"
+
+using namespace std;
+using namespace ccf;
+using namespace crypto;
+using namespace tls;
+
+namespace ccfapp
+{
+
+
+    bool TPHandler::verify_sig_static(const vector<uint8_t>& signature, const tls::PublicKeyPtr & pubk_verifier, \
```

Comment:
Can't we enforce that all verifying keys must be passed in as strings as in `verify_sig()`? Alternatively, the one verification function that calls this function directly could just call `pubk_verifier->verify()` directly if you'd rather not change the interface. Just seems odd to have a function for a single-line operation that is used once throughout the code.

Reply:
thank you for the question. the verify_sig_static will be used at least twice (one as you noted above, one for verifying enclave signature for the add_enclave_transaction. This verification is cuurrently broken, so the method returns true, I am working on fixing this. Yes, the function definition is only one line now, but that should not concern us. 

verify_sig_static cannot be subsituted with verify_sig for performance reasons. I agree the function name can perhaps be improved. happy to take suggestions.